### PR TITLE
update bedgraphtobigwig with additional platform

### PR DESCRIPTION
**Pull request by pabloaledo**
add additional platform to compile for aarch.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/ucsc-bedgraphtobigwig/meta.yaml [hunk 1]
```diff
@@ -12,9 +12,10 @@ source:
   sha256: "{{ sha256 }}"
   patches:
     - "include.patch"
+    - "pathc1.patch"
```
Comment (Joon-Klaps) [inline recipes/ucsc-bedgraphtobigwig/meta.yaml:15]:
```suggestion
    - "patch1.patch"
```

File: recipes/ucsc-bedgraphtobigwig/patch1.patch [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+@@ -711,11 +711,11 @@
+ puts("Status: 400\r");
+ puts("Content-Type: text/plain; charset=UTF-8\r");
+ puts("\r");
+-if (format != NULL && args != NULL)
+-    {
+-    vfprintf(stdout, format, args);
+-    fprintf(stdout, "\n");
+-    }
++if (format != NULL)
```
Comment (martin-g) [inline recipes/ucsc-bedgraphtobigwig/patch1.patch:12]: This has been fixed upstream several months ago - https://github.com/ucscGenomeBrowser/kent/commit/8b044625d9cdd9dfaec89ff32d0ecb069b7c0565
This fix is in tag `[v463_preview](https://github.com/ucscGenomeBrowser/kent/releases/tag/v463_preview)`

Comment (martin-g) [reply recipes/ucsc-bedgraphtobigwig/patch1.patch:12]: Changing the sources would be a big change indeed! Let's continue with the patch for now!
Please apply the suggestion - https://github.com/bioconda/bioconda-recipes/pull/46516#discussion_r1529823216

**Comment by martin-g** [comment]
```

mgrigorov in 🌐 euler-arm-22 in /tmp/bedgraphtobigwig via 🅒 font-ubuntu took 3s 
❯ tar jxvf ucsc-bedgraphtobigwig-455-hd8417b0_1.tar.bz2                                                                                                                                                (font-ubuntu) 
info/files
info/run_exports.json
info/recipe/run_test.sh
info/recipe/357/run_test.sh
info/test/run_test.sh
info/has_prefix
info/hash_input.json
info/recipe/patch1.patch
info/index.json
info/recipe/357/include.patch
info/recipe/357/build.sh
info/paths.json
info/recipe/build.sh
info/recipe/meta.yaml.template
info/recipe/include.patch
info/recipe/357/meta.yaml
info/recipe/meta.yaml
info/recipe/conda_build_config.yaml
info/about.json
info/git
bin/bedGraphToBigWig

mgrigorov in 🌐 euler-arm-22 in /tmp/bedgraphtobigwig via 🅒 font-ubuntu 
❯ file bin/bedGraphToBigWig                                                                                                                                                                            (font-ubuntu) 
bin/bedGraphToBigWig: ELF 64-bit LSB pie executable, ARM aarch64, version 1 (SYSV), dynamically linked, interpreter /lib/ld-linux-aarch64.so.1, for GNU/Linux 3.7.0, stripped

mgrigorov in 🌐 euler-arm-22 in /tmp/bedgraphtobigwig via 🅒 font-ubuntu 
❯ ./bin/bedGraphToBigWig --help                                                                                                                                                                        (font-ubuntu) 
./bin/bedGraphToBigWig: error while loading shared libraries: libmysqlclient.so.18: cannot open shared object file: No such file or directory

mgrigorov in 🌐 euler-arm-22 in /tmp/bedgraphtobigwig via 🅒 font-ubuntu 
❯ conda install ucsc-bedgraphtobigwig-455-hd8417b0_1.tar.bz2                                                                                                                                           (font-ubuntu) 

Downloading and Extracting Packages
                                                                                                                                                                                                                     

Downloading and Extracting Packages

Preparing transaction: done
Verifying transaction: done
Executing transaction: done

mgrigorov in 🌐 euler-arm-22 in /tmp/bedgraphtobigwig via 🅒 font-ubuntu took 3s 
❯ bedGraphToBigWig                                                                                                                                                                                     (font-ubuntu) 
bedGraphToBigWig: error while loading shared libraries: libmysqlclient.so.18: cannot open shared object file: No such file or directory

mgrigorov in 🌐 euler-arm-22 in /tmp/bedgraphtobigwig via 🅒 font-ubuntu 
❯ conda install mysql-connector-c                                                                                                                                                                      (font-ubuntu) 
Collecting package metadata (current_repodata.json): done
Solving environment: / 
The environment is inconsistent, please check the package plan carefully
The following packages are causing the inconsistency:

  - <unknown>/linux-aarch64::ucsc-bedgraphtobigwig==455=hd8417b0_1
done

## Package Plan ##

  environment location: /home/mgrigorov/miniconda3/envs/font-ubuntu

  added / updated specs:
    - mysql-connector-c


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    libpng-1.6.43              |       h194ca79_0         287 KB  conda-forge
    mysql-connector-c-6.1.11   |    h8945da3_1008         1.3 MB  conda-forge
    ------------------------------------------------------------
                                           Total:         1.6 MB

The following NEW packages will be INSTALLED:

  libpng             conda-forge/linux-aarch64::libpng-1.6.43-h194ca79_0 
  mysql-connector-c  conda-forge/linux-aarch64::mysql-connector-c-6.1.11-h8945da3_1008 


Proceed ([y]/n)? y


Downloading and Extracting Packages
                                                                                                                                                                                                                     
Preparing transaction: done                                                                                                                                                                                          
Verifying transaction: done
Executing transaction: done

mgrigorov in 🌐 euler-arm-22 in /tmp/bedgraphtobigwig via 🅒 font-ubuntu took 35s 
❯ bedGraphToBigWig                                                                                                                                                                                     (font-ubuntu) 
bedGraphToBigWig v 2.10 - Convert a bedGraph file to bigWig format (bbi version: 4).
usage:
   bedGraphToBigWig in.bedGraph chrom.sizes out.bw
where in.bedGraph is a four column file in the format:
      <chrom> <start> <end> <value>
and chrom.sizes is a two-column file/URL: <chromosome name> <size in bases>
and out.bw is the output indexed big wig file.
If the assembly <db> is hosted by UCSC, chrom.sizes can be a URL like
  http://hgdownload.soe.ucsc.edu/goldenPath/<db>/bigZips/<db>.chrom.sizes
or you may use the script fetchChromSizes to download the chrom.sizes file.
If not hosted by UCSC, a chrom.sizes file can be generated by running
twoBitInfo on the assembly .2bit file.
The input bedGraph file must be sorted, use the unix sort command:
  LC_ALL=C sort -k1,1 -k2,2n unsorted.bedGraph > sorted.bedGraph
The LC_ALL=C variable activates case-sensitive sorting.
options:
   -blockSize=N - Number of items to bundle in r-tree.  Default 256
   -itemsPerSlot=N - Number of data points bundled at lowest level. Default 1024
   -sizesIsBb  -- If set, the chrom.sizes file is assumed to be a bigBed file.
   -unc - If set, do not use compression.
```

Looks good to me!

**Comment by martin-g** [comment]
@BiocondaBot please add label

**Review by rpetit3 — APPROVED** [review]

**Pull request merged**